### PR TITLE
Updated Parser code 

### LINKED ISSUE #18: Ravi's Ability to Greet the user
The application Ravi should be able to greet the user, allowing them to know that it has been successfully executed and running.

### LINKED ISSUE #19: Ravi's ability to bid farewell to the user
The application Ravi should be able to bid farewell to the user, allowing them to know that it has been successfully terminated

### LINKED ISSUE #20: Ravi's ability to differentiate the user's input
The application Ravi is required to make sense of the user's input command and execute the necessary functions to have the desired output

### PR DESCRIPTION
**Pull request by tobiasceg**
Changed the switch statement 
Added sean's method for addcommand 

EditTaskCommand was added to remove errors

### CODE REVIEW

**Comment by wangqinNick** [comment]
Looks quite good. If it passes the checks, then can merge.

**Comment by Aseanseen** [comment]
Looks ok, but the Parser needs more prepare commands (delete, list, done)

**Comment by SimJunYou** [comment]
Checks seem to be failing due to missing portions from other parts of the code base. Parser code seems good to me.

File: src/main/java/seedu/duke/parser/Parser.java [hunk 1]
```diff
@@ -2,17 +2,16 @@
...
 import java.security.InvalidParameterException;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 import static seedu.duke.util.ExceptionMessage.MESSAGE_INVALID_PARAMETERS;
-import static seedu.duke.util.Message.MESSAGE_CHECK_COMMAND_FORMAT;
-import static seedu.duke.util.Message.MESSAGE_EMPTY_INPUT;
-import static seedu.duke.util.Message.MESSAGE_INVALID_COMMAND_FORMAT;
-import static seedu.duke.util.Message.MESSAGE_NO_EDIT_MODULE;
+import static seedu.duke.util.Message.*;
```
Comment (Aseanseen) [inline src/main/java/seedu/duke/parser/Parser.java]: Violates coding standard due to *

File: src/main/java/seedu/duke/parser/Parser.java [hunk 2]
```diff
@@ -56,19 +72,47 @@ public Command parseCommand(String input) {
             return new IncorrectCommand(MESSAGE_INVALID_COMMAND_FORMAT);
         }
         String commandWord = matcher.group(COMMAND_WORD_GROUP).toLowerCase().trim();
+        String commandFlag = matcher.group(COMMAND_FLAG_GROUP).toLowerCase();
         String parameters = matcher.group(PARAMETERS_GROUP);
 
         try {
             switch (commandWord){
```
Comment (Aseanseen) [inline src/main/java/seedu/duke/parser/Parser.java]: This method seems a bit long, maybe do more abstraction

**Review by Aseanseen — COMMENTED** [review]
Few edits, but overall looks fine

File: src/main/java/seedu/duke/parser/Parser.java (hunk 2 above)
Comment (tobiasceg) [reply src/main/java/seedu/duke/parser/Parser.java]: i dont think i can abstract this switch statement due to the fact that it returns different methods based on the command_word_****? Any advice?

File: src/main/java/seedu/duke/parser/Parser.java [hunk 3]
```diff
@@ -56,19 +79,58 @@ public Command parseCommand(String input) {
...
             }
         } catch (InvalidParameterException e) {
             return new IncorrectCommand(MESSAGE_INVALID_PARAMETERS);
         }
+        return null;
+    }
+
+    private DeleteCommand getDeleteCommand(String commandFlag, String parameters) {
+        if (commandFlag.equals(MODULE_PREFIX)) {
+            return new DeleteCommand(typeOfEntries.MODULE, null , parameters); //parameter is module code
```
Comment (Aseanseen) [inline src/main/java/seedu/duke/parser/Parser.java]: Need to remove the null due to updated constructor for DeleteCommand. 1 more occurence below.

**Review by Aseanseen — CHANGES_REQUESTED** [review]
Looks ok, minor edits

File: src/main/java/seedu/duke/parser/Parser.java [hunk 4]
```diff
@@ -100,6 +174,47 @@ protected Command prepareEditModuleCommand(String parameters)
...
+            if (taskDeadline.isEmpty()) { // -by is present but empty deadline
+                return new IncorrectCommand(String.format("%s%s\n\n%s%s\n",
+                      MESSAGE_INVALID_COMMAND_FORMAT, taskDeadline, MESSAGE_CHECK_COMMAND_FORMAT, AddCommand.FORMAT));
+            }
+        }
+
+        // no task input by user
+        if (isNothingToEdit(addedTask)) {
+            return (typeOfTask == TypeOfEntries.MODULE)
+                    ? new IncorrectCommand(MESSAGE_NO_ADD_MODULE) : new IncorrectCommand(MESSAGE_NO_ADD_TASK);
```
Comment (Aseanseen) [inline src/main/java/seedu/duke/parser/Parser.java]: Violates coding standard for ternary expressions

File: src/main/java/seedu/duke/util/ExceptionMessage.java [hunk 5]
```diff
@@ -1,6 +1,6 @@
 package seedu.duke.util;
 
-public class ExceptionMessage {
+public class    ExceptionMessage {
```
Comment (Aseanseen) [inline src/main/java/seedu/duke/util/ExceptionMessage.java]: Unsure why the spaces?

**Review by Aseanseen — CHANGES_REQUESTED** [review]
Need some amends, thanks

File: src/main/java/seedu/duke/ui/TextUi.java [hunk 6]
```diff
@@ -1,24 +1,125 @@
...
+    /**
+     * Formats an Arraylist of type Task with their Index.
+     *
+     * @param taskList the list to be formatted
+     */
+    public static String getIndexTaskList(ArrayList<Task> taskList) {
+        final StringBuilder stringFormat = new StringBuilder();
+        int displayIndex = 0 + DISPLAY_INDEX_OFFSET;
+        for (Task t : taskList) {
+            stringFormat.append(getIndexListFormat(displayIndex, t.toString())).append("\n");
```
Comment (Aseanseen) [inline src/main/java/seedu/duke/ui/TextUi.java:68]: Not sure about the readability of this line

**Pull request merged**